internal/scm/bitbucketcloud/bitbucket_cloud.go

```diff
@@ -0,0 +1,324 @@
+package bitbucketcloud
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"slices"
+	"strings"
+
+	"github.com/ktrysmt/go-bitbucket"
+	"github.com/lindell/multi-gitter/internal/scm"
+	"github.com/pkg/errors"
+)
+
+type BitbucketCloud struct {
+	repositories []string
+	workspaces   []string
+	users        []string
+	username     string
+	token        string
+	sshAuth      bool
+	httpClient   *http.Client
+	bbClient     *bitbucket.Client
+}
+
+func New(username string, token string, repositories []string, workspaces []string, users []string, sshAuth bool, transportMiddleware func(http.RoundTripper) http.RoundTripper) (*BitbucketCloud, error) {
+	if strings.TrimSpace(token) == "" {
+		return nil, errors.New("bearer token is empty")
+	}
+
+	bitbucketCloud := &BitbucketCloud{}
+	bitbucketCloud.repositories = repositories
+	bitbucketCloud.workspaces = workspaces
+	bitbucketCloud.users = users
+	bitbucketCloud.username = username
+	bitbucketCloud.token = token
+	bitbucketCloud.sshAuth = sshAuth
+	bitbucketCloud.httpClient = &http.Client{
+		Transport: transportMiddleware(&http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: false}, // nolint: gosec
+		}),
+	}
+	bitbucketCloud.bbClient = bitbucket.NewBasicAuth(username, token)
+
+	return bitbucketCloud, nil
+}
+
+func (bbc *BitbucketCloud) CreatePullRequest(ctx context.Context, repo scm.Repository, prRepo scm.Repository, newPR scm.NewPullRequest) (scm.PullRequest, error) {
+
+	splitRepoFullName := strings.Split(prRepo.FullName(), "/")
```

Comment:
Could you calrify what you mean by `type asserting` here?

Reply:
`prRepo.(repository)` is type asserting to the repository struct. Since it's guaranteed to be of that type, we can do it. The important part is to make it clearer

`splitRepoFullName[1]` is hard to understand without first parsing the surrounding code.
`bbcPrRepo.name` is self explanatory 😄

Reply:
Type assertions came in handy and we were mostly able to use them.